File: fiskalizimi/Signer.cs
using System.Security.Cryptography;

namespace fiskalizimi;

public interface ISigner
{
    // returns base64 encoded signature of the signed bytes
    string SignBytes(byte[] data);
}

public class Signer : ISigner
{
    private readonly string _key;
    
    public Signer(string pemKey)
    {
        _key = pemKey;
    }
    
    public string SignBytes(byte[] data)
    {
        var ecdsa = ECDsa.Create();
        ecdsa.ImportFromPem(_key);

        using (var sha256 = SHA256.Create())
        {
            var hash = sha256.ComputeHash(data);
            var signature = ecdsa.SignHash(hash);
            
            var encodedSignature = Convert.ToBase64String(signature);
            return encodedSignature;
        }
    }
}

File: fiskalizimi/Signer.php
<?php

namespace fiskalizimi;

interface ISigner
{
    // Returns base64 encoded signature of the signed bytes
    public function signBytes(string $data): string;
}

class Signer implements ISigner
{
    private string $key;

    public function __construct(string $pemKey)
    {
        $this->key = $pemKey;
    }

    public function signBytes(string $data): string
    {
        // Load the private key
        $privateKey = openssl_pkey_get_private($this->key);
        if (!$privateKey) {
            throw new \Exception('Error loading private key.');
        }

        // Hash the data using SHA256
        $hash = hash('sha256', $data, true);

        // Sign the hash
        $signature = '';
        $success = openssl_sign($hash, $signature, $privateKey, OPENSSL_ALGO_SHA256);
        if (!$success) {
            throw new \Exception('Error signing data.');
        }

        // Free the private key resource
        openssl_free_key($privateKey);

        // Return the base64-encoded signature
        return base64_encode($signature);
    }
}